etc/testing/test_tls.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+set -euo pipefail
+
+pachctl version
+# Don't log our activation code when running this script in Travis
+pachctl enterprise activate "$(aws s3 cp s3://pachyderm-engineering/test_enterprise_activation_code.txt -)" && echo
+
+set -x
+
+# Split PACHD_ADDRESS into host and port
+IFS=: read -a addr_parts <<<"${PACHD_ADDRESS:-0.0.0.0:30650}"
+unset PACHD_ADDRESS # use config value set by gen_pachd_tls.sh
```

Comment:
Where does this get set in the first place?

`PACHD_ADDRESS` will be removed at some point, so it might be worth just tackling that now.

Reply:
I use it when testing locally :joy: . Travis uses the default value. Would you be okay with leaving it in until I'm forced to find a new way to work locally?

I way I could imagine seeing this work longer term is to add a pachctl command like `pachctl address` or some such that spits out the pachyderm address that it's connected to, and then having this script use that to avoid having to copy the address-determining logic that's already in `client.go`. How would you feel about taking that approach, but doing it in another PR?

Reply:
Yeah makes sense to wait.

I'm not sure if I see much utility to a `pachctl address`, but this is may very well be lack of familiarity with parts of the testing infrastructure.

Reply:
Also, for posterity, I'm wrong about `PACHD_ADDRESS` being definitely removed at some point. It's not marked for deprecation yet, though it may be in the future. Given how disruptive its removal would be, though, it's safe to say that it's at least going to be around for the foreseeable future.

The reasoning is that contexts provide an alternative, and, while `PACHD_ADDRESS` is convenient for some usecases, it's also dangerous: it makes it trivial to send one cluster's credentials to another cluster.

Reply:
That makes sense.

I don't feel any particular urgency to make this change, but I guess in case we decide to come back to it later, my idea with `pachctl address` was that there's a block of reasonably complex logic in `client.go` (in [getUserMachineAddrAndOps](https://github.com/pachyderm/pachyderm/blob/master/src/client/client.go#L300)) that determines which address `pachctl` is currently connected to. If the purpose of `gen_pachd_tls.sh` is to generate a self-signed TLS cert for your current pach cluster, then it could either duplicate the logic in `getUserMachineAddrAndOps` or ask `pachctl` to execute that logic and print the result. `pachctl address` would do the latter.

There are a lot of alternatives we could do, though—I don't feel at all wedded to this one. We could have `pachctl` generate the self-signed cert, or decide that we don't like self-signed certs and use some other PKI for setting up TLS